### PR TITLE
feat: add azure files integration

### PR DESCRIPTION
**Pull request by PIC123**
A previous pr added dynamic volume creation based on the volumes in the benchmark. These changes add support for doing that with azure files. Adds a storage class and a reference to the storage class in the generated workflow

### CODE REVIEW

**Review by tessmichi — APPROVED** [review]

File: deploy/helm/values.dev.yaml [hunk 1]
```diff
@@ -18,6 +18,8 @@ worker:
   image: acanthamoeba/sds-worker
   queueMode: azure
   queueEndpoint: https://<storageaccount>.queue.core.windows.net/runs
+  resourceGroup: resourceGroup
```
Comment (noelbundick) [inline deploy/helm/values.dev.yaml:21]: This is really the resource group for the entire SDS deployment, not specific to the worker. Can we bump this up a level?

File: deploy/helm/values.dev.yaml [hunk 2]
```diff
@@ -18,6 +18,8 @@ worker:
...
+  storageAccount: storageAccount
```
Comment (noelbundick) [inline deploy/helm/values.dev.yaml:22]: Can we pick a more descriptive name? At a glance, it's confusing why this is "storageAccount", but then queueEndpoint is also "storageaccount". We don't want people to pick the same one here - they have distinct uses & access patterns. `runsTempStorageAccount` or something seems like a better fit

**Review by noelbundick — CHANGES_REQUESTED** [review]
I see there are no new integration tests to validate this configuration. Have you validated it to work both in a dev environment and a locked-down (PrivateLink) AKS cluster? Do we have any results / is there any way to automate that validation?

File: deploy/helm/templates/azurefiles-csi.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: azurefile-csi
+provisioner: file.csi.azure.com
```
[the remark this answers is not in the document]
Comment (noelbundick) [reply deploy/helm/templates/azurefiles-csi.yaml:5]: See also: https://docs.microsoft.com/en-us/azure/aks/csi-storage-drivers

File: deploy/helm/values.dev.yaml (hunk 2 above)
Comment (PIC123) [reply deploy/helm/values.dev.yaml:22]: That works. Just curious though, why would we not be able to use the same storage account. We'll just be creating new file shares and then deleting them post run so seemed like it wouldn't be too much of an issue using the same one

Comment (noelbundick) [reply deploy/helm/values.dev.yaml:22]: The queue storage acct is network dual-homed - it's allowed to be accessed inside the detonation chamber for the worker, but also outside the detonation chamber by the laboratory. This temp share is purely internal and sometimes holds a copy of the data we're trying to protect during processing. We don't want that exposed outside in any way

**Pull request closed without merging**